test/topology_custom/test_mv_backlog.py

```diff
@@ -76,3 +81,113 @@ async def test_gossip_same_backlog(manager: ManagerClient) -> None:
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Only remote updates hold on to memory, so make the update remote
+    srv_base = servers[0]
+    srv_view = servers[1]
+    host_base = next(h for h in hosts if h.address == srv_base.ip_addr)
+    await pin_the_only_tablet(manager, "ks", "tab", srv_base)
+    await pin_the_only_tablet(manager, "ks", "mv_cf_view", srv_view)
+
+    # All nodes in the cluster run with --smp=1, so there is only shard 0
+    shard = 0
+
+    delay_metric_name = 'scylla_storage_proxy_coordinator_mv_flow_control_delay_total'
+    throttled_writes_metric_name = 'scylla_storage_proxy_coordinator_throttled_base_writes_total'
+
+    delay_limits = [0, 500, 1000, 2000, 10000]
+    computed_delays = []
+
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+
+    for delay_limit in delay_limits:
+        logger.info(f"delay_limit = {delay_limit}")
+
+        # Update the delay
+        await asyncio.gather(*(cql.run_async(f"UPDATE system.config SET value = '{delay_limit}' WHERE name = 'view_flow_control_delay_limit_in_ms'", host=h) for h in hosts))
+
+        # Make sure that view updates will hang
+        await asyncio.gather(*(manager.api.enable_injection(s.ip_addr, "never_finish_remote_view_updates", one_shot=False) for s in servers))
+
+        # Measure the total delay before the write, and the number of delayed writes
+        local_metrics = await manager.metrics.query(srv_base.ip_addr)
+        before_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+        before_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+
+        # Generate a large view update and then a small one.
+        # The reason why we do two writes is as follows: view backlog is propagated
+        # in responses from base writes and the coordinator caches it but it will
+        # not necessarily use it when calculating the delay of the same write.
+        # The second small write will use the value of the backlog from the previous write.
+        await cql.run_async(stmt, [0, 0, 100000*'a'], host=host_base)
+        await cql.run_async(stmt, [0, 0, ''], host=host_base)
+
+        # Make sure that there is exactly one throttled write and calculate a delay for it.
+        # If we're testing the 0ms delay, instead make sure that there were no delayed writes.
+        local_metrics = await manager.metrics.query(srv_base.ip_addr)
+        after_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+        after_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+
+        if delay_limit == 0:
+            assert after_total_throttled_writes == before_total_throttled_writes
+        else:
+            assert after_total_throttled_writes >= before_total_throttled_writes + 1
```

Comment:
In the comment above we say that we make sure there's only one throttled write, but here we assert that there's at least one throttled write. We should update the comment or the assert so that they match

Reply:
Good point. However, instead of doing that I changed the test in a different way - I measure the count/delay before and after the second write only, instead of including both writes. This way, if there is some change to how view backpressure is propagated and the first write starts being throttled, too, the test won't break.